mlx/backend/metal/device.cpp

```diff
@@ -19,16 +19,14 @@ namespace mlx::core::metal {
 
 namespace {
 
-// Catch things related to the main-thread static variables
-static std::shared_ptr<void> global_memory_pool = new_scoped_memory_pool();
-
 // TODO nicer way to set this or possibly expose as an environment variable
 static constexpr int MAX_BUFFERS_PER_QUEUE = 12;
 
 static constexpr const char* default_mtllib_path = METAL_PATH;
 
 auto load_device() {
-  MTL::Device* device = MTL::CreateSystemDefaultDevice();
+  auto devices = MTL::CopyAllDevices();
+  auto device = static_cast<MTL::Device*>(devices->object(0));
```

Comment:
not strictly autoreleasepool related, but observed with command line and xctest runs: MTL::CreateSystemDefaultDevice is not available to all process types.

Reply:
> MTL::CreateSystemDefaultDevice is not available to all process types.

Interesting. I wonder if this will fix #14.

Reply:
No need for a pool there? I remember seeing an unsolved leak regarding some MTL initialization

Reply:
(but in a thread we do not control, so I remember)

Reply:
Yes, there is some activity done via `dispatch_async()` from Metal itself that we can't catch.

This pool was removed because a new one was added in `void Device::new_queue(int index) {`. This allows us to remove the pool from the scheduler's `thread_fn()` -- the CPU scheduler didn't need it and the metal scheduler already had it via the `task()`.

This was per offline discussion with @awni